### DISCUSSION TITLE
PrimeFaces Maven Repository down?

### DISCUSSION THREAD

**Discussion by s0ftwar3punK** [primefaces/community#2382] [PrimeFaces] [answered]
```xml
<repositories>
        <repository>
            <id>primefaces</id>
            <name>PrimeFaces Maven Repository</name>
            <url>http://repository.primefaces.org</url>
            <layout>default</layout>
        </repository>
</repositories>
```

**Comment by melloware** [comment]
What is your question?   The PrimeFaces Maven Repo?

**Comment by haarli** [comment]
I suppose this is about the PrimeFaces MavenRepo. We have similar issues, the old themes dependencies org.primefaces.themes:all-themes:jar:1.0.10 and org.primefaces.themes:themes-project:pom:1.0.10 are not available anymore since yesterday: 
https://repository.primefaces.org/org/primefaces/themes/ gives a 404 NOT FOUND.

Was this removed on purpose? Any other repository where this is still downloadable?

Thank you

**Reply by melloware** [reply]
cc @cagataycivici @mertsincan  ???

**Reply by s0ftwar3punK** [reply]
Somehow my question is gone, but yes it was about this topic and we have exactly this problem.

**Reply by dgisbert** [reply]
Having the same issue. Seems that themes repository has disappeared

**Comment by melloware** [comment]
Can you guys post what you needed that repo for?  Whatever it is we should put in Maven Central so you no longer need that repo?

**Reply by pdurbin** [reply]
@melloware hi, thanks for asking!

@haarli and I are working together so I'll answer for both of us. I'm pretty sure these are the only artifacts we need:

```
./org/primefaces/themes/all-themes/1.0.10/all-themes-1.0.10.jar
./org/primefaces/themes/all-themes/1.0.10/all-themes-1.0.10.jar.sha1
./org/primefaces/themes/all-themes/1.0.10/all-themes-1.0.10.pom
./org/primefaces/themes/all-themes/1.0.10/all-themes-1.0.10.pom.lastUpdated
./org/primefaces/themes/all-themes/1.0.10/all-themes-1.0.10.pom.sha1
./org/primefaces/themes/all-themes/1.0.10/all-themes-1.0.10.jar.lastUpdated
./org/primefaces/themes/themes-project/1.0.10/themes-project-1.0.10.pom.lastUpdated
./org/primefaces/themes/themes-project/1.0.10/themes-project-1.0.10.pom
./org/primefaces/themes/themes-project/1.0.10/themes-project-1.0.10.pom.sha1
```

Putting these on Maven Central would be ideal! Thanks!

**Reply by melloware** [reply, marked as answer]
OK its on Maven Central now: https://repo1.maven.org/maven2/org/primefaces/themes/all-themes/1.0.11/

You should be able to use the same maven coordinates

```xml
<dependency>
   <groupId>org.primefaces.themes</groupId>
   <artifactId>all-themes</artifactId>
   <version>1.0.11</version>
</dependency>
```

If you used Barcode4J fork from PF Repo that is also on Maven Central here:

```xml
<dependency>
    <groupId>org.primefaces.extensions</groupId>
    <artifactId>barcode4j-light</artifactId>
    <version>2.3.0</version>
</dependency>
```

**Reply by dgisbert** [reply]
Thank you very much. It's too much to ask having each theme separately as before ?

We use 

```
<dependency>
	<groupId>org.primefaces.themes</groupId>
	<artifactId>bluesky</artifactId>
	<version>1.0.10</version>
</dependency>

<dependency>
	<groupId>org.primefaces.themes</groupId>
	<artifactId>bootstrap</artifactId>
	<version>1.0.10</version>
</dependency>
```

**Reply by melloware** [reply]
@codasven @dgisbert its a pain for me to put 40 more jars up on Maven Central I recommend you just switch to the all-themes JAR.  Its only 1MB and if this was 20 years ago increasing your WAR size by 1MB would be a bigger deal...but not in 2024 today's modern web.

**Reply by melloware** [reply]
@codasven @dgisbert i just uploaded 1.0.11 to Maven Central including all the individual theme jars.  It should be there within the hour.

```xml
<dependency>
	<groupId>org.primefaces.themes</groupId>
	<artifactId>bluesky</artifactId>
	<version>1.0.11</version>
</dependency>

<dependency>
	<groupId>org.primefaces.themes</groupId>
	<artifactId>bootstrap</artifactId>
	<version>1.0.11</version>
</dependency>
```

**Comment by pdurbin** [comment]
@melloware thanks for uploading all-themes to Maven Central! [Problem solved](https://github.com/IQSS/dataverse/issues/10718) for me and @haarli. ❤️

**Comment by Noenzbj** [comment]
@melloware Is it possible to upload the 1.0.8 version aswell?

**Reply by Noenzbj** [reply]
we only need org.primefaces.themes:bootstrap:jar:1.0.8

**Reply by melloware** [reply]
Isn't bootstrap in 1.0.10 all themes?

**Reply by Noenzbj** [reply]
yes but we need it in 1.0.8, would be great :)

**Reply by melloware** [reply]
I don't understand. 1.0.10 is a newer version of Boostrap why would you want an older 1.0.8 version?

**Reply by Noenzbj** [reply]
yeah, i totaly agree with you but our project depends on 1.0.8. 🙈 
if it's too much trouble then please don't do it. we'll just have to update it to 1.0.10 then and hope everything will work :D

**Reply by melloware** [reply]
It will work you realize both 1.0.8 and 1.0.10 have not been updated in over 10 years 😄  i think you will be fine.

**Reply by Noenzbj** [reply]
From your lips to God's ears. 😄 thank you anyway for your time to answer :)

**Reply by Noenzbj** [reply]
i tried everything i got from the internet but it still doesnt work. I get the "[Error loading theme, cannot find "theme.css" resource of primefaces theme library](https://stackoverflow.com/questions/29025709/error-loading-theme-cannot-find-theme-css-resource-of-primefaces-theme-librar)". In some forums they recomend to downgrade to 1.0.8 bcs 1.0.10 does not work properly. Our project is sadly that old and we are working on it to write it in springboot.

**Reply by melloware** [reply]
@Noenzbj not sure what you are doing but try this:
[legacy-bootstrap.zip](https://github.com/user-attachments/files/16426953/legacy-bootstrap.zip)

Unzip and run `mvn clean jetty:run` and navigate to http://localhost:8080/ you will see bootstrap works just fine.
